tests/pytest/tests.sh

```diff
@@ -90,32 +121,115 @@ valgrind_summary() {
 
 run_tests() {
 	local title="$1"
-	[[ ! -z $title ]] && { $ROOT/opt/readies/bin/sep -0; printf "Tests with $title:\n\n"; }
+	shift
+	if [[ -n $title ]]; then
+		$READIES/bin/sep -0
+		printf "Testing $title:\n\n"
+	fi
+
 	cd $ROOT/tests/pytest
-	[[ ! -z $TESTMOD ]] && RLTEST_ARGS+=--module $TESTMOD
-	$OP python3 -m RLTest --clear-logs --module $MODULE --module-args "JSON_BACKEND SERDE_JSON" $RLTEST_ARGS
-	$OP python3 -m RLTest --clear-logs --module $MODULE $RLTEST_ARGS
+
+	[[ $SERDE_JSON == 1 ]] && MODARGS+=" JSON_BACKEND SERDE_JSON"
+	
+	if [[ $EXISTING_ENV != 1 ]]; then
+		rltest_config=$(mktemp "${TMPDIR:-/tmp}/rltest.XXXXXXX")
+		rm -f $rltest_config
+		cat <<-EOF > $rltest_config
+			--oss-redis-path=$REDIS_SERVER
+			--module $MODULE
+			--module-args '$MODARGS'
+			$RLTEST_ARGS
+			$VALGRIND_ARGS
+			$SAN_ARGS
+			$COV_ARGS
+
+			EOF
+
+	else # existing env
+		xredis_conf=$(mktemp "${TMPDIR:-/tmp}/xredis_conf.XXXXXXX")
+		rm -f $xredis_conf
+		cat <<-EOF > $xredis_conf
+			loadmodule $MODULE $MODARGS
+			EOF
+
+		rltest_config=$(mktemp "${TMPDIR:-/tmp}/xredis_rltest.XXXXXXX")
+		rm -f $rltest_config
```

Comment:
Same - if calling mktemp you don't need to remove the file, even though bashwarting (cat <<-EOF) to the new file.